File: C++/LeetCode/src/TwoSum/twosum.cpp
#include <iostream>
#include <ostream>
#include <vector>
#include <map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> map;
        for (int i = 0; i < nums.size(); i++) {
            auto it = map.find(target - nums[i]);
            if (it != map.end()) {
                return {it, i};
            }
            map[nums[i]] = i;
        }
        return {};
    }
};

int main() {
    Solution sol = Solution();
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    vector<int> res = sol.twoSum(nums, target);
    cout << "[" << res[0] << ", " << res[1] << "]" << endl;
    return 0;
}

File: Java/src/LeetCode/TwoSum/TwoSum.java
package LeetCode.TwoSum;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class TwoSum {
	int[] twoSum(int[] nums, int target) {
		Map<Integer, Integer> map = new HashMap<>();
		for (int i = 0; i < nums.length; i++) {
			int left = target - nums[i];
			if (map.containsKey(left)) {
				return new int[] { map.get(left), i };
			}
			map.put(nums[i], i);
		}
		return null;
	}

	public static void main(String[] args) {
		TwoSum lzr = new TwoSum();
		int[] nums = { 1, 2, 3 };
		int target = 5;
		int[] res = lzr.twoSum(nums, target);
		System.out.print(Arrays.toString(res));
	}
}